.github/workflows/canary-integration-test.yml

```
@@ -1265,6 +1265,61 @@ jobs:
+        run: |
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume create myfs testsubvolume
+
+      - name: Create subvolume of same name on secondary cluster
+        run: |
+          kubectl exec -n rook-ceph-secondary deploy/rook-ceph-tools -t -- ceph fs subvolume create myfs testsubvolume
+
+      - name: Deploy Direct Tools pod on primary cluster
+        run: |
+          kubectl -n rook-ceph create -f deploy/examples/direct-mount.yaml
+
+      - name: Deploy Direct Tools pod on secondary cluster
+        run: |
+          sed -i "s/rook-ceph # namespace/rook-ceph-secondary # namespace/" deploy/examples/direct-mount.yaml
+          kubectl -n rook-ceph-secondary create -f deploy/examples/direct-mount.yaml
+
+      - name: Configure a directory for snapshot mirroring on primary cluster
+        run: |
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs snapshot mirror enable myfs
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs snapshot mirror add myfs /volumes/_nogroup/testsubvolume/
+
+      - name: make sure that snapshot mirror is enabled on the secondary cluster
+        run: |
+          kubectl exec -n rook-ceph-secondary deploy/rook-ceph-tools -t -- ceph fs snapshot mirror enable myfs
+
+      - name: Create 3 snapshots on cluster primary cluster
+        run: |
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume snapshot create myfs testsubvolume snap1
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume snapshot create myfs testsubvolume snap2
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume snapshot create myfs testsubvolume snap3
+
+      - name: Get the peer and verify the peer synchronization status that snaps have synced on secondary cluster
+        run: |
+          exec_fs_mirror='kubectl -n rook-ceph exec deploy/rook-ceph-fs-mirror --'
+          mirror_daemon=$($exec_fs_mirror ls /var/run/ceph/ | grep "fs-mirror" | head -n 1)
+          timeout 45 bash -x <<EOF
+            while
+            clusterfsid=\$($exec_fs_mirror ceph --admin-daemon /var/run/ceph/$mirror_daemon fs mirror status myfs@1 |jq -r '.peers|keys[]')
+            [ -z "\$clusterfsid" ]
+            do echo "Waiting for the clusterfsid to get populated." && sleep 1
+            done
+          EOF
+          clusterfsid=$($exec_fs_mirror ceph --admin-daemon /var/run/ceph/$mirror_daemon fs mirror status myfs@1 |jq -r '.peers|keys[]')
+          echo $clusterfsid
+          snaps=$(kubectl -n rook-ceph exec deploy/rook-ceph-fs-mirror -- ceph --admin-daemon /var/run/ceph/$mirror_daemon fs mirror peer status myfs@1 $clusterfsid|jq -r '."/volumes/_nogroup/testsubvolume"."snaps_synced"')
+          echo $snaps
+          #timeout 15 sh -c 'while [ -z "$clusterfsid" ]; do echo "Waiting for the clusterfsid to get populated." && sleep 1; done'
```

```suggestion
          #timeout 15 sh -c 'while [ -z "$clusterfsid" ]; do echo "Waiting for the clusterfsid to get populated." && sleep 1; done'
```

this is a comment right, do you want to remove maybe add `echo` if you want, I see the timeout below so we can remove this?

---

@subhamkrai ah yep i intend it to be removed.